### ISSUE TITLE
renaming / deleting non-existing file fails without notifying the user

### ISSUE THREAD

**Issue by individual-it** [owncloud/web#1720]
### Steps to reproduce
1. create a file
2. with an other client delete the file
3. try to rename or delete the file in phoenix (phoenix does not have the information that the file does not exist any-more)

### Expected behaviour
user should be notified
when trying to overwrite the file the user gets at least `File upload failed… Precondition failed `
when trying to share the file the user gets `Error while sharing. Wrong path, file/folder doesn't exist `
something similar should happen when renaming or deleting a non-existing resource

### Actual behaviour
no feedback to the user

### Server configuration
**Operating system**:
Ubuntu 18.04
**Web server:**
Apache 2
**Database:**
MySQL
**PHP version:**
PHP 7.1
**ownCloud version:** (see ownCloud admin page)
10.3.0 prealpha (git) 
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git
**The content of config/config.php:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "172.17.0.1"
        ],
        "datadirectory": "\/home\/artur\/www\/owncloud-core\/data",
        "overwrite.cli.url": "http:\/\/localhost\/owncloud-core\/",
        "version": "10.3.0.0",
        "logtimezone": "UTC",
        "htaccess.RewriteBase": "\/owncloud-core\/",
        "ldapIgnoreNamingRules": false,
        "loglevel": 1,
        "log_type": "owncloud",
        "dbname": "owncloud_20190701",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "sharing.federation.allowHttpFallback": true,
        "files_external_allow_create_new_local": "true",
        "license-key": "***REMOVED SENSITIVE VALUE***",
        "instanceid": "oc4kyqahfanl",
        "apps_paths": [
            {
                "path": "\/home\/artur\/www\/owncloud-core\/apps",
                "url": "apps",
                "writable": true
            },
            {
                "path": "\/home\/artur\/www\/owncloud-core\/apps2",
                "url": "apps2",
                "writable": true
            }
        ],
        "dbtype": "mysql",
        "allow_user_to_change_display_name": false,
        "csrf.disabled": true,
        "enable_previews": false,
        "maintenance": false,
        "cors.allowed-domains": [
            "http:\/\/172.17.0.1:8300"
        ],
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "skeletondirectory": "\/home\/artur\/www\/owncloud-core\/apps\/testing\/data\/webUISkeleton"
    }
}

```

**List of activated apps:**

```
Enabled:
  - activ#### Web server error logity: 2.4.2
  - comments: 0.3.0
  - dav: 0.4.0
  - diagnostics: 0.1.3
  - enterprise_key: 0.1.4
  - federatedfilesharing: 0.5.0
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_lifecycle: 1.0.0
  - files_sharing: 0.12.0
  - files_texteditor: 2.3.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firstrunwizard: 1.2.0
  - notifications: 0.5.0
  - oauth2: 0.4.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - systemtags_management: 0.3.1
  - testing: 0.1.0
  - updatenotification: 0.2.1
  - user_management: 0.0.1
  - windows_network_drive: 0.7.2
  - workflow: 0.3.0
Disabled:
  - admin_audit
  - brute_force_protection
  - customgroups
  - encryption
  - files_antivirus
  - files_classifier
  - files_primary_s3
  - firewall
  - guests
  - password_policy
  - ransomware_protection
  - richdocuments
  - search_elastic
  - sharepoint
  - twofactor_totp
  - user_ldap
  - wopi
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
FF 68
**Operating system:**
Ubuntu 18.04

#### Browser log
renaming:
```
Unhandled promise rejection Error: "File with name a-folder/a1.txt could not be located"
    r http://172.17.0.1:8300/core/core.bundle.js:69
    r http://172.17.0.1:8300/core/core.bundle.js:69
    e http://172.17.0.1:8300/core/core.bundle.js:69
    value http://172.17.0.1:8300/core/core.bundle.js:68
    value http://172.17.0.1:8300/core/core.bundle.js:69
    J http://172.17.0.1:8300/apps/files/files.bundle.js:7
    r http://172.17.0.1:8300/core/core.bundle.js:61
files.bundle.js:15:98639
    exports http://172.17.0.1:8300/apps/files/files.bundle.js:15
    q http://172.17.0.1:8300/apps/files/files.bundle.js:7
    n http://172.17.0.1:8300/apps/files/files.bundle.js:7
    exports http://172.17.0.1:8300/apps/files/files.bundle.js:15
    Q http://172.17.0.1:8300/apps/files/files.bundle.js:7
    <anonymous> http://172.17.0.1:8300/core/core.bundle.js:55
    b http://172.17.0.1:8300/core/core.bundle.js:55
    x http://172.17.0.1:8300/core/core.bundle.js:55
```

deleting:
```
error: /a-folder/a1-renamed.txt not deleted: Error: File with name a-folder/a1-renamed.txt could not be located
```

**Comment by individual-it** [comment]
similar problem with other operations: download / fav / un-fav

**Comment by DeepDiver1975** [comment]
Maybe already take care of - some reject handling has been worked on these days ..... needs reevaluation

**Comment by individual-it** [comment]
most cases are taken care of but "favorite" and "unfavorite" are still problematic, see tests in #2528

**Comment by individual-it** [comment]
took off assignment and QA team label because tests are implemented in #2528

**Comment by fschade** [comment]
testet in lates web version with oCis backend, web now takes care of it and shows a notification.
![Uploading Bildschirmfoto 2022-03-24 um 14.53.16.png…]()
